src/item_search.cpp

```diff
@@ -46,9 +74,9 @@ item_filter_from_string(  std::string filter )
             filter = filter.substr( colon + 1 );
         }
     }
-    bool exclude = filter[0] == '-';
+    bool exclude = filter.find( '-' ) != std::string::npos;
```

Comment:
Any particular reason for the change?
Actually, you check for negation twice (if exclude duplicates): here and above (when separating functions) which means you do the same work twice: you can either drop [this block](https://github.com/CleverRaven/Cataclysm-DDA/pull/18933/files#diff-2ab5d626787b18b8d3cbabb28d4243d6R80) and `return !f( i );` or leave the functional negation and find a more elegant solution.